src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -130,7 +130,7 @@ export class SharedCredentialsProvider implements CredentialsProvider {
             }
             const provider = new AWS.CredentialProviderChain([this.makeCredentialsProvider()])
 
-            return await provider.resolvePromise()
+            return (await resolveProviderWithCancel(this.profileName, provider.resolvePromise())) as AWS.Credentials
```

Comment:
Unnecessary `as AWS.Credentials`?

Reply:
Yeah not needed anymore. Was there before due to mismatched type definitions.